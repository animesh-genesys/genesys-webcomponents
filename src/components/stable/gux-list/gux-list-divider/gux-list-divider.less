@import (reference) '../../../../style/color.less';

:host {
  position: relative;
  display: flex;
  width: 100%;
  height: 1px;
  padding: 0;
  margin: 16px 0;
  line-height: 32px;
  pointer-events: none;
  cursor: pointer;
  cursor: default;
  background-color: @gux-grey-40;
}
