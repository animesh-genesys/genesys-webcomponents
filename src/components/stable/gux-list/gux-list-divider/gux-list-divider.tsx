import { Component, h, Host, JSX } from '@stencil/core';

@Component({
  styleUrl: 'gux-list-divider.less',
  tag: 'gux-list-divider',
  shadow: true
})
export class GuxListDivider {
  render(): JSX.Element {
    return (<Host role="presentation"></Host>) as JSX.Element;
  }
}
