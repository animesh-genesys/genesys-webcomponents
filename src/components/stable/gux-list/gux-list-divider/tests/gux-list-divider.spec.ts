import { newSpecPage } from '@stencil/core/testing';
import { GuxListDivider } from '../gux-list-divider';

const components = [GuxListDivider];
const language = 'en';

describe('gux-list-divider', () => {
  it('should build', async () => {
    const page = await newSpecPage({
      components,
      html: `<gux-list-divider></gux-list-divider>`,
      language
    });

    expect(page.rootInstance).toBeInstanceOf(GuxListDivider);
  });

  describe('#render', () => {
    ['<gux-list-divider></gux-list-divider>'].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
