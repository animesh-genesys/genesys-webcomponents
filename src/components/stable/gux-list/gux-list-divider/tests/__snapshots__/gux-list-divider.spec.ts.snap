// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-list-divider #render should render component as expected (1) 1`] = `
<gux-list-divider role="presentation">
  <mock:shadow-root></mock:shadow-root>
</gux-list-divider>
`;
